api/provider/aws/load_balancer_delete.go

```diff
@@ -21,6 +26,33 @@ func (l *LoadBalancerProvider) Delete(loadBalancerID string) error {
 		return err
 	}
 
+	// Check for eventually consistency
+	fnWaitTillLBDeleted := func() (shouldRetry bool, err error) {
+		input := &elb.DescribeLoadBalancersInput{}
+		input.SetLoadBalancerNames([]*string{aws.String(fqLoadBalancerID)})
+		input.SetPageSize(1)
+
+		output, err := l.AWS.ELB.DescribeLoadBalancers(input)
+		if err != nil {
+			if err, ok := err.(awserr.Error); ok && err.Code() == "LoadBalancerNotFound" {
+				return false, nil
+			}
+
+			return false, err
+		}
+
+		if len(output.LoadBalancerDescriptions) != 1 {
```

Comment:
I think this should be `== 0`; it's unlikely, but possible that the len > 1

Reply:
This code path can never be reached as you're checking for a `LoadBalancerNotFound` error and returning in line 37. Also the pagesize is set to 1 so len should never in theory be greater than 1. But in practice it'll only ever be 1 if it gets to here, if i'm reading this code correctly so you shouldn't need this.